src/test/suite/storage/connectionStorage.test.ts

```diff
@@ -321,6 +322,134 @@ suite('Connection Storage Test Suite', function () {
       extensionSandbox.restore();
     });
 
+    suite('when there are preset connections', () => {
+      const presetConnections = {
+        globalValue: [
+          {
+            name: 'Global Connection 1',
+            connectionString: 'mongodb://localhost:27017/',
+          },
+        ],
+        workspaceValue: [
+          {
+            name: 'Preset Connection 1',
+            connectionString: 'mongodb://localhost:27017/',
+          },
+          {
+            name: 'Preset Connection 2',
+            connectionString: 'mongodb://localhost:27018/',
+          },
+        ],
+      };
+
+      let getConfigurationStub: sinon.SinonStub<
+        [
+          section?: string | undefined,
+          scope?: vscode.ConfigurationScope | null | undefined
+        ],
+        vscode.WorkspaceConfiguration
+      >;
+      let inspectPresetSavedConnectionsStub: sinon.SinonStub;
+
+      beforeEach(() => {
+        testSandbox.restore();
+        inspectPresetSavedConnectionsStub = testSandbox.stub();
+      });
+
+      test('loads the preset connections', async () => {
+        getConfigurationStub = testSandbox.stub(
+          vscode.workspace,
+          'getConfiguration'
+        );
+        getConfigurationStub.returns({
+          inspect: inspectPresetSavedConnectionsStub,
+          get: () => undefined,
+        } as any);
+
+        inspectPresetSavedConnectionsStub
+          .withArgs('presetSavedConnections')
+          .returns(presetConnections);
+
+        const loadedConnections = await testConnectionStorage.loadConnections();
+
+        const expectedConnectionValues = [
+          ...presetConnections.globalValue.map((connection) => ({
+            ...connection,
+            source: 'globalSettings',
+          })),
+          ...presetConnections.workspaceValue.map((connection) => ({
+            ...connection,
+            source: 'workspaceSettings',
+          })),
+        ];
+
+        expect(loadedConnections.length).equals(
+          expectedConnectionValues.length
+        );
+
+        for (let i = 0; i < expectedConnectionValues.length; i++) {
+          const connection = loadedConnections[i];
+          const expected = expectedConnectionValues[i];
+          expect(connection.name).equals(expected.name);
+          expect(connection.connectionOptions.connectionString).contains(
```

Comment:
q: why are we checking for contains here rather than equals?

Reply:
ah actually it was because the connectionString ends up losing the trailing `/`, will still adjust it to be stricter and more defined